File: Interpreter/Ruby/AndExpression.rb
require_relative "IExpression"

class AndExpression < IExpression
    def initialize(expression1, expression2)
        @expression1 = expression1
        @expression2 = expression2
    end

    def solve(context)
        if !context.include? "&&"
            return false
        end

        return @expression1.solve(context) && @expression2.solve(context)
    end
end

File: Interpreter/Javascript/AndExpression.js
import IExpression from "./IExpression.js";

class AndExpression extends IExpression {
  constructor(expression1, expression2) {
    super();
    this.expression1 = expression1;
    this.expression2 = expression2;
  }

  solve(context) {
    if (!context.includes("&&")) {
      return false;
    }

    return this.expression1.solve(context) && this.expression2.solve(context);
  }
}

export default AndExpression;